test/loopback-dl.test.js

```diff
@@ -1473,6 +1483,20 @@ describe('DataAccessObject', function() {
     assert(error, 'An error should have been thrown');
   });
 
+  it('should throw when malformed logical operators follow valid logical clauses', function() {
```

Comment:
@doublemarked unfortunately in many places tests start with `should` and we should eventually make them consistent. For now as a team we decided to follow [these documented rules](https://github.com/strongloop/loopback.io/blob/gh-pages/pages/en/contrib/style-guide.md#test-naming) where it says: 
```
Use imperative mood, do not start test names with should.
```

Lets follow them.

Thanks :-)